engine/graphics/src/vulkan/graphics_vulkan_context.cpp

```diff
@@ -191,30 +207,29 @@ namespace dmGraphics
         }
 
         int32_t enabled_layer_count = 0;
+        dmArray<VkExtensionProperties> extensions;
+        GetExtensions(extensions);
+
+        // Static to keep life time while adding it to an array and passing it along
+        static const char* VK_KHR_get_physical_device_properties2_str = "VK_KHR_get_physical_device_properties2";
 
         if (validationLayerCount > 0)
         {
             if (GetValidationSupport(validationLayers, validationLayerCount))
             {
                 enabled_layer_count = validationLayerCount;
 
-            #ifdef __MACH__
-                vk_required_extensions.OffsetCapacity(2);
-                vk_required_extensions.Push(VK_KHR_PORTABILITY_ENUMERATION_EXTENSION_NAME);
-                vk_required_extensions.Push("VK_KHR_get_physical_device_properties2");
-            #else
-                vk_required_extensions.OffsetCapacity(1);
-                vk_required_extensions.Push(VK_KHR_PORTABILITY_ENUMERATION_EXTENSION_NAME);
-            #endif
+                AddIfSupported(extensions, VK_KHR_PORTABILITY_ENUMERATION_EXTENSION_NAME, vk_required_extensions);
+                AddIfSupported(extensions, VK_KHR_get_physical_device_properties2_str, vk_required_extensions);
```

Comment:
One of them weren't supported on Switch (I don't recall which one)

Reply:
Yeah, the vulkan version on switch is probably a bit behind. Next time we need validation for switch I'll make sure to use the correct validation layers, but for now it's more important it works on osx/win32